### PR TITLE
Keycloak integration

### LINKED ISSUE #2: Keycloak / OIDC
Our authentication should append in [Keycloak](https://www.keycloak.org) via OIDC. A request that needs authentication, should be redirected by our api-gateway (to-be-discussed). For our local development, Keycloak should not be required. We should rather be able to generate development jwts to develop services independent of Keycloak.

The production configuration for Keycloak should be defined via Iac.

### PR DESCRIPTION
**Pull request by FoseFx**
Fixes #2

 - [x] add custom keycloak image
 - [x] add keycloak auth
 - [x] update user-svc to use keycloak
 - [x] write helmfiles
 - [x] use custom image
 - [x] import realm (iac)

I'll leave it at that, but there is quite a lot of work to be done.

When reviewing it's probably best to do so an an per-commit basis, I rebased and organized it a bit.

### CODE REVIEW

**Review by MaxSchaefer — COMMENTED** [review]
Puh, large pr. I skimmed it, looks good so far. Question: Can't Keycloak take care of all the account creation? So the auth-svc would not even get in touch with the credentials.

File: kubernetes/keycloak-operator/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Based on https://www.keycloak.org/operator/installation
```
[the remark this answers is not in the document]
Comment (FoseFx) [reply kubernetes/keycloak-operator/README.md:1]: you mean existing helm chart? Yes, a couple. But most of the one's I found on artifacthub had not enough values to set or very low usage. None of them are on a stable version, so may change and cause headaches later on. That's why I wrote the chart myself (based on the yaml files provided by keycloak).

**Comment by FoseFx** [comment]
> Can't Keycloak take care of all the account creation? So the auth-svc would not even get in touch with the credentials.

I thought about it, here are a couple of reasons I opted against this:
- main thing I had in mind were non-keycloak releated operations that might be needed after creating a new user. We would need to hook into keycloak's event system (which is, I think, only meant for logging).
- We have to rely on KC's UI. Sure we can theme it to our liking, but I'm not sure how much control we have about certain elements (e.g., complex date widgets or whatnot)
- Ideally I want to take away the permissions for self management for users in keycloak and only allow changes though us. This way we stay ahead of invalid data. I am also no KC expert, and want to have as little magic as possible going on in our arguably most critical part of the whole system.

**Comment by MaxSchaefer** [comment]
> > Can't Keycloak take care of all the account creation? So the auth-svc would not even get in touch with the credentials.
> 
> I thought about it, here are a couple of reasons I opted against this:
> 
> * main thing I had in mind were non-keycloak releated operations that might be needed after creating a new user. We would need to hook into keycloak's event system (which is, I think, only meant for logging).
> * We have to rely on KC's UI. Sure we can theme it to our liking, but I'm not sure how much control we have about certain elements (e.g., complex date widgets or whatnot)
> * Ideally I want to take away the permissions for self management for users in keycloak and only allow changes though us. This way we stay ahead of invalid data. I am also no KC expert, and want to have as little magic as possible going on in our arguably most critical part of the whole system.

Good points, sounds reasonable. Thank you!

**Review by MaxSchaefer — APPROVED** [review]
🚀🌊

**Pull request merged**